### ISSUE TITLE
Do you plan to support @MonotonicNonNullDecl

### ISSUE THREAD

**Issue by chengniansun** [uber/NullAway#93]
Do you plan to support @MonotonicNonNullDecl? Recently I added this annotation to guava (https://github.com/google/guava/blob/master/guava/src/com/google/common/reflect/TypeToken.java#L105)

Or more generally, do you support the nullness annotations of the checker framework?

**Comment by msridhar** [comment]
We can easily support interpreting the CF annotations as best as possible within the (less sophisticated) NullAway type system.  Actually enhancing the power of the type system would be more work and would require some further justification in terms of additional reliability in practice, etc.

For this one, I think by default we would just treat the field as `@NonNull`.  Are you looking for deeper support than that?

**Comment by msridhar** [comment]
Ah, I see.  You're probably looking to not see reports of initialization errors on such fields, yes?  If so, that can easily be handled using the `-XepOpt:NullAway:ExcludedFieldAnnotations` option (see [here](https://github.com/uber/NullAway/wiki/Configuration)).

**Comment by chengniansun** [comment]
For this one, should we treat it in the following way, 

If a field $f$ annotated with @monotonicnonnulldecl is not initialized in a constructor, NullAway does not emit a warning saying that " initializer method does not guarantee @NonNull fields f is initialized along all control-flow paths"

I am currently focusing on eliminating all warnings on uninitialized fields.

**Comment by msridhar** [comment]
Yeah, for now you can handle this by passing `-XepOpt:NullAway:ExcludedFieldAnnotations=MonotonicNonNullDecl` as an Error Prone / javac argument.  For CF compatibility we should probably bake in support for this annotation.

**Comment by chengniansun** [comment]
Neat. Thank you.
